src/test/ui/entry-point/imported_main_const_fn_item_type_forbidden.stderr

```diff
@@ -21,7 +21,7 @@ LL |     type MainFn = impl Fn();
 LL |     pub const BAR: MainFn = bar;
    |                             ^^^ expected opaque type, found fn item
    |
-   = note: expected opaque type `impl Fn<()>`
+   = note: expected opaque type `impl Fn()`
                   found fn item `fn() {bar}`
```

Comment:
(not for this PR): These cases should have an additional `note` explaining the difference between `fn` and `Fn`, and what you can do to deal with that.